### PR TITLE
Newer spark

### PR DESCRIPTION
**Pull request by cgore**

### CODE REVIEW

File: src/main/clojure/powderkeg/core.clj [hunk 1]
```diff
@@ -247,7 +247,7 @@
         df (barrier-fn (fn [it] (eduction (comp may-unmap-tuple2 xform may-map-tuple2) (iterator-seq it))))
         rdd (.mapPartitions rdd
               (reify org.apache.spark.api.java.function.FlatMapFunction ; todo: skip api.java.* go to spark
-                (call [_ it] ((df) it)))
+                (call [_ it] (.iterator ((df) it))))
```
Comment (cgrand) [inline src/main/clojure/powderkeg/core.clj]: the return type of call changed from Iterator to Iterable...

Comment (viesti) [reply src/main/clojure/powderkeg/core.clj]: Hmm, seems so https://spark.apache.org/docs/2.1.0/api/java/org/apache/spark/api/java/function/FlatMapFunction.html vs 1.5.0. Should we do a different change here? Above only avoids ClassCastException.

Comment (cgrand) [reply src/main/clojure/powderkeg/core.clj]: it was more of a "note to self" to remember why this change

Comment (viesti) [reply src/main/clojure/powderkeg/core.clj]: Yep, seems a bit odd to me too, would assume that many things in Spark itself would have changed and this change somewhat stands out and looks odd :)

File: src/main/clojure/powderkeg/core.clj [hunk 2]
```diff
@@ -401,7 +401,7 @@
         rdds (map ensure-scala-pair-rdd rdds)
         partitioner (org.apache.spark.Partitioner/defaultPartitioner
                       rdd (scala-seq rdds))]
-    (by-key (org.apache.spark.rdd.CoGroupedRDD. (scala-seq (cons rdd rdds)) partitioner)
+    (by-key (org.apache.spark.rdd.CoGroupedRDD. ^scala.reflect.ClassTag (scala-seq (cons rdd rdds)) partitioner)
```
Comment (cgrand) [inline src/main/clojure/powderkeg/core.clj]: Not in the right place, and has been preserved in subsequent commits.

Comment (viesti) [reply src/main/clojure/powderkeg/core.clj]: Hmm, true. Not sure how to create the evidence parameter actually, but I see your on it in spark2 branch :)

**Comment by cgrand** [comment]
I merged into the spark2 branch, made it compatible with 1.5 and added lein profiles.
Please test it with profiles spark1.5 and spark2.
If it's ok for you I'll merge in `master`.

**Pull request merged**